### DISCUSSION TITLE
La Crosse View sensors

### DISCUSSION THREAD

**Discussion by derobert** [merbanan/rtl_433#2169] [Ideas]
tldr: I'm thinking of trying to add a unified La Crosse View device, based on the sensor ID to distinguish.

I've got a bunch of LTV-WL1 sensors, so am about to write some code based on the other La Crosse View devices — seems there are four so far (Breeze Pro, R1, TH3, WR1).

I'm pretty sure they all actually are the same protocol. For example, the WiFi-connected stations happily receive from any of the sensors & forward to their cloud service, even if they can't display the sensor data locally. https://www.lacrossetechnology.com/pages/view-sensor-compatibility-us for example has a list; almost without exception everything has the `*` meaning it can pass through to the cloud.

`-X n=LCV,m=FSK_PCM,s=104,l=104,r=9600,preamble={24}aa2dd4` seems to collect all of them (at least the W1, TH2, TH3, and WL1.)

Right now it looks like all the different devices distinguish mainly by packet length. But... even from the same sensor, that's variable. [Bitbench WL1, after preamble](https://triq.net/bitbench#c=%7B156%7D50b964040f02643100000000000000000000000&c=%7B155%7D505ea70c0f02511e00000000000000000000000&c=%7B156%7D50b9de040f02519f00000000000000000000000&c=%7B153%7D504936800f02587800000000000000000000000&c=%7B155%7D50b964060f02643600000000000000000000000&c=%7B155%7D50b9c4020f0259fa00000000000000000000000&c=%7B156%7D505ea70e0f02511900000000000000000000000&c=%7B153%7D50b9de060f02519800000000000000000000000&c=%7B156%7D504936820f02587f00000000000000000000000&c=%7B156%7D50b964080f02642300000000000000000000000&c=%7B156%7D50b9c4040f0259f300000000000000000000000&c=%7B248%7D505ea7000f02510c0000000000000000000000000000000000000000000000&c=%7B156%7D50b9de080f02518d00000000000000000000000&c=%7B249%7D504936840f02594700000000000000000000000000000000000000000000000&c=%7B154%7D50b9640a0f02642400000000000000000000000&c=%7B154%7D50b9c4060f0259f400000000000000000000000&c=%7B251%7D505ea7020f02510b00000000000000000000000000000000000000000000000&c=%7B156%7D50b9de0a0f0252d900000000000000000000000&c=%7B150%7D504936860f0259400000000000000000000000&c=%7B156%7D50b9640c0f02642d00000000000000000000000&c=%7B156%7D50b9c4080f0259e100000000000000000000000&c=%7B250%7D505ea7040f02525100000000000000000000000000000000000000000000000&c=%7B152%7D50b9de0c0f0252d00000000000000000000000&c=%7B203%7D504936880f02595500000000000000000000000000000000000&c=%7B155%7D50b9640e0f02642a00000000000000000000000&c=%7B155%7D50b9c40a0f0259e600000000000000000000000&c=%7B156%7D504936000f025aeb00000000000000000000000%20%5Bnew%20batteries%5D&c=%7B155%7D505ea7060f02525600000000000000000000000&c=%7B193%7D50b9de0e0f0252d7000000000000000000000000000000000&c=%7B156%7D50b964000f02643f00000000000000000000000&f=ID%3A24h%20BATTLOW%3A1b%20%3F%3A3b%20SEQ%3A3d%20%3F%3A1b%20WETDRY%3A12h%20TEMP%3A12d%20CRC%3A8h%20512x&cw=4)

Here is some with the pre-amble, including testing adding water to one, which changes the wetdry field from 0f0 to f0f. I also confirmed it uses the same (raw-400)/10 formula: [Bitbench with preamble](https://triq.net/bitbench#c=%7B173%7Daaaaaaaaaaaaaab4aa8b7514124da103c09651c00000%20%20%20%5B68.2F%20dry%20lowbat%5D&c=%7B167%7Daaaaaaaaaaaaaab4aa8b7514124da183c096500000%20%20%20%5B68.2F%20dry%20lowbat%5D&c=%7B218%7Daaaaaaaaaaaaaab4aa8b751417a9c103c0944080000000000000000%20%20%5B66.7F%20dry%20good_bat%5D&c=%7B199%7Daaaaaaaaaaaaaab4aa8b751417a9c183c09441400000000000%20%5B66.7F%20dry%20good_bat%5D&c=%7B180%7Daaaaaaaaaaaaaab4aa8b75142e590003c099438000000%20%5B70.3F%2021.3C%20dry%20good%20bat%5D&c=%7B171%7Daaaaaaaaaaaaaab4aa8b75142e778103c094b300000%20%5B19.4C%20good%20bat%20dry%5D&c=%7B173%7Daaaaaaaaaaaaaab4aa8b75142e7781bc3c94ea400000%20%5B19.5C%20wet%5D&c=%7B170%7Daaaaaaaaaaaaaab4aa8b75142e778203c0951e00000%20%5B19.6C%20dry&c=%7B171%7Daaaaaaaaaaaaaab4aa8b75142e778283c095cb00000%20%5B19.9C%20dry%5D&f=ID%3A24h%20BATTLOW%3A1b%20STARTUP%3A1b%20%3F%3A2b%20SEQ%3A3h%20%3F%3A1b%20WETDRY%3A12h%20TEMP%3A12d%20CRC%3A8h%20TRAILER%208h8h8h8h8h8h8h8h&a=Preamble&m=aa2dd4&cw=4)

[Note that still had the on startup bit in the format, but I confirmed when replacing the battery that the WL1 sensor didn't set that. Also, I haven't done the CRC math yet to make sure it's actually a CRC.]

So, err, I sort of think the only way to tell the sensors apart is by the device ID. I notice all my sensors of the same type start with the same hex digit... I have a bunch of TH2, they all start with 3 (from 3207d9 through 37c6e6); the WL1 sensors all start 50 (504936 through 50b9de). I've actually got more of those on order, so probably that will be a higher ID.

(The WL1 appears to use the same packet format as TH2, just with different meanings for the 2 12-bit fields. And oddly they didn't put the temperature in the same spot.)

**Comment by zuckschwerdt** [comment]
CRC checks out to the usual poly 0x31 init 0x00.

Re ID/type: is the ID printed somewhere or could the first nibble/byte be just a type code really? We see that on other protocols.

*edit:* just noted the "ID: 3 bytes (matches bar code underside of unit covering pgm port)" comment in BreezePro. The first bits could be a type code still.

**Reply by derobert** [reply]
Yeah, they all have the matching ID printed on them (barcoded sticker). And the same ID is in the Android app. But yeah, I think they're using some number of bits to mark the sensor type, since my IDs all cluster by the sensor type (and I didn't buy all the TH2s or WL1s at once).

I left that `-X` running for a few hours today, and I've got two sensors it hasn't detected — a TH2 (but it does detect the other 8 TH2s) and a R2 rain gauge. It might just be a signal strength issue; I need to try it on a laptop closer to those sensors.
